lib/bundler/process_lock.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Bundler
+  class ProcessLock
+    def self.lock(bundle_path = Bundler.bundle_path)
+      lock_file_path = File.join(bundle_path, "bundler.lock")
+      File.open(lock_file_path, "w") do |f|
+        f.flock(File::LOCK_EX)
```

Comment:
OK I am going to have to dig further on this one, I had assumed it was well supported and could not find anything in the contrary either looking at the ruby source or docs.